### PR TITLE
Adds a workaround for dragleave event in safari

### PR DESCRIPTION
**Pull request by meg1502**
Fixes: SIRI-666

### CODE REVIEW

File: src/main/resources/default/assets/common/core.js.pasta [hunk 1]
```diff
@@ -489,3 +489,31 @@ sirius.writeLocalStore = function (name, value) {
...
+        return true;
+    }
+    if (event.clientY >= window.innerHeight) {
+        return true;
+    }
+    if (event.clientY <= 1) {
+        return true;
+    }
+    return false;
+}
```
Comment (jakobvogel) [inline src/main/resources/default/assets/common/core.js.pasta:519]:
```suggestion
}

```

File: src/main/resources/default/assets/common/core.js.pasta [hunk 2]
```diff
@@ -489,3 +489,31 @@ sirius.writeLocalStore = function (name, value) {
...
+    if (event.clientX >= window.innerWidth) {
+        return true;
+    }
+    if (event.offsetX <= 0) {
+        return true;
+    }
+    if (event.clientY >= window.innerHeight) {
+        return true;
+    }
+    if (event.clientY <= 1) {
```
Comment (jakobvogel) [inline src/main/resources/default/assets/common/core.js.pasta:515]: why not `< 0`?

File: src/main/resources/default/assets/common/core.js.pasta [hunk 3]
```diff
@@ -489,3 +489,31 @@ sirius.writeLocalStore = function (name, value) {
...
+sirius.isDragleaveEventLeavingWindow = function (event) {
+    const isSafari = /^((?!chrome|android|edge).)*safari/i.test(navigator.userAgent);
+
+    if (!isSafari && event.relatedTarget === null) {
+        return true;
+    }
+    if (event.clientX >= window.innerWidth) {
+        return true;
+    }
+    if (event.offsetX <= 0) {
```
Comment (jakobvogel) [inline src/main/resources/default/assets/common/core.js.pasta:509]: why not `< 0`?

**Review by jakobvogel — APPROVED** [review]

**Comment by jakobvogel** [comment]
Does that work when dragging quickly?

**Review by sabieber — APPROVED** [review]

**Pull request merged**